src/main/kotlin/org/rust/lang/refactoring/RustLocalVariableHandler.kt

```diff
@@ -0,0 +1,204 @@
+package org.rust.lang.refactoring
+
+import com.intellij.codeInsight.PsiEquivalenceUtil
+import com.intellij.openapi.actionSystem.DataContext
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Pass
+import com.intellij.psi.*
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.refactoring.IntroduceTargetChooser
+import com.intellij.refactoring.RefactoringActionHandler
+import com.intellij.refactoring.introduce.inplace.InplaceVariableIntroducer
+import com.intellij.refactoring.introduce.inplace.OccurrencesChooser
+import org.rust.lang.core.psi.*
+import java.util.*
+
+class RustLocalVariableHandler : RefactoringActionHandler {
+    override fun invoke(project: Project, editor: Editor?, file: PsiFile?, dataContext: DataContext?) {
```

Comment:
Sweet, this is how intelij autogenerated the signature for me, but removing it makes it a lot simpler, awesome